expression/builtin_miscellaneous_test.go

```diff
@@ -55,8 +56,11 @@ func TestInetAton(t *testing.T) {
 		f, err := fc.getFunction(ctx, datumsToConstants(tt["Input"]))
 		require.NoError(t, err)
 		d, err := evalBuiltinFunc(f, chunk.Row{})
-		require.NoError(t, err)
-		testutil.DatumEqual(t, tt["Expected"][0], d)
+		if err == nil {
```

Comment:
Your fix for tests are not correct. All these fixes should be fixed somehow.

You should do something like:

```
if tt.expecterr !=  nil {
        require.True(t, terror.ErrorEqual(err, tt.expecterr))
} else {
        require.NoError(t, err)
		testutil.DatumEqual(t, tt["Expected"][0], d)
}
```

In short, the result of execution must be fail/succ. If the implementation is wrong and tje exeution failed, won't it also get `errWrongValueForType`? Then how does `require.True(t, terror.ErrorEqual(err, errWrongValueForType))` make sense? Your fix of tests break some fundamental logic of testing.

Reply:
Hi, thks much for comment. but I think the tt..expecterr sure is errWrongValueForType, so I use what really output to compare with it should be OK, like many other test's logic. e.g. tt["Expected"][0] is just set by self on the above map code. We can also set errWrongValueForType message in the  tt map but that make the code look bad, so I did this. is this right?

Reply:
> so I did this. is this right?

No, obviously.

> tt..expecterr sure is errWrongValueForType

Yes, but the point is that, it is the case written by ourselves determines whether the test should fail or not. Not the result of the test determines whether it should fail or not. How could use the execution result to determine if it should fail? What are you testing? Did you get the poit?

The correct fix may be add a new column to indicate whether the test should fail or not. Or use something like `_, ok := tt["Expected"][0].(error)` to indicate whether the test should fail or not. Or whatever other methods.

Reply:
unterstood, you are right, I'll do some amendation and recommit. THKS.